### PR TITLE
USER02 - Allow a User to register and login to the application

### PR DESCRIPTION
**Pull request by GiorgioCampea**
Created the component LoginComponent to let the user signin or signup using Firebase-UI

The component is on the index page:
if the user is not logged:
localhost_4321_ (2)

if the user is logged:
localhost_4321_ (4)

There is a known issue for email provider for switching from signup to signin

https://github.com/firebase/firebaseui-web/issues/1040#issuecomment-1772045964

disable the email enumeration to make it work

### CODE REVIEW

File: src/components/LoginComponent.astro [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+<script>
+  import { inMemoryPersistence } from "firebase/auth";
+  import firebase from "firebase/compat/app";
+  import * as firebaseui from "firebaseui";
+  import "firebaseui/dist/firebaseui.css";
+  import { auth } from "../firebase/client";
+
+  const uiConfig = {
+    signInOptions: [
+      // Leave the lines as is for the providers you want to offer your users.
```
Comment (gregoriopalama) [inline src/components/LoginComponent.astro:14]: remove the comments

File: src/components/LoginComponent.astro [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      // google, apple, email
+      firebase.auth.GoogleAuthProvider.PROVIDER_ID,
+      // in order to make the switch from signin to signup for email 
+      // remember to disable the the email enumeration protection
+      // https://github.com/firebase/firebaseui-web/issues/1040
```
Comment (gregoriopalama) [inline src/components/LoginComponent.astro:19]: remove these comments

File: src/components/LoginComponent.astro [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      firebase.auth.EmailAuthProvider.PROVIDER_ID,
+      'apple.com'
+    ],
+    // tosUrl and privacyPolicyUrl accept either url string or a callback
+    // function.
+    // Terms of service url/callback.
+    tosUrl: "<your-tos-url>",
```
Comment (gregoriopalama) [inline src/components/LoginComponent.astro:26]: remove the comments and add a tos url. I would add a very simple page, for now, with just a "Terms of service". we will add an issue to work on it

File: src/components/LoginComponent.astro [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    // Privacy policy url/callback.
+    privacyPolicyUrl: function () {
+      window.location.assign("<your-privacy-policy-url>");
```
Comment (gregoriopalama) [inline src/components/LoginComponent.astro:29]: same for the tos url

File: src/components/LoginComponent.astro [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      signInSuccessWithAuthResult: function(authResult: any, redirectUrl?: string) {
+        return false;
+      },
+      signInFailure: function(error: firebaseui.auth.AuthUIError) {
+        // Some unrecoverable error occurred during sign-in.
+        // Return a promise when error handling is completed and FirebaseUI
+        // will reset, clearing any UI. This commonly occurs for error code
+        // 'firebaseui/anonymous-upgrade-merge-conflict' when merge conflict
+        // occurs. Check below for more details on this.
+        return console.error(error);
```
Comment (gregoriopalama) [inline src/components/LoginComponent.astro:41]: remove the comments

**Review by gregoriopalama — CHANGES_REQUESTED** [review]

**Comment by GiorgioCampea** [comment]
some important comment, useful for documentation:

- in order to make the switch from signin to signup for email remember to disable the the email enumeration protection [see this github issue](https://github.com/firebase/firebaseui-web/issues/1040)
- all the javascript in astro is bundled, so we need to check if the starting div is present [see this github issue](https://github.com/withastro/astro/issues/5739)

**Comment by irsooti** [comment]
I would like to remark that firebase-ui comes with a [stylesheet](https://github.com/firebase/firebaseui-web/blob/master/stylesheet/firebase-ui.css)which doesn't stick on our style guide - _At this point, we are in a hurry, and could skip this aspect for now._

An upcoming enhancement could involve customizing buttons and fonts according to our style guide, as opposed to using the default firebase-ui design.

**Comment by gregoriopalama** [comment]
> I would like to remark that firebase-ui comes with a [stylesheet](https://github.com/firebase/firebaseui-web/blob/master/stylesheet/firebase-ui.css)which doesn't stick on our style guide - _At this point, we are in a hurry, and could skip this aspect for now._
> 
> An upcoming enhancement could involve customizing buttons and fonts according to our style guide, as opposed to using the default firebase-ui design.

I totally agree on this.
It's not just that we are in a hurry. Other issues need this functionality, so the quickest and dirtiest way sounds good for us, for now. 
When we will work on the public UI, we will surely have to fix this

**Review by gregoriopalama — APPROVED** [review]

**Review by irsooti — APPROVED** [review]

**Pull request merged**